### ISSUE TITLE
LocationEngineProvider crash when obtainBestLocationEngineAvailable()

### ISSUE THREAD

**Issue by Niduank** [mapbox/mapbox-events-android#132]
Hi guys,

I follow your example to get location but app crash.

Device : 
Samsung Galaxy S8+
Android : 8.0.0

My code : 
```java
locationEngine = new LocationEngineProvider(context).obtainBestLocationEngineAvailable();
// Same crash if I do locationEngine = new LocationEngineProvider(context).obtainLocationEngineBy(LocationEngine.Type.GOOGLE_PLAY_SERVICES);
locationEngine.activate();
locationEngine.addLocationEngineListener(this);
```

There is my gradle : 
```
implementation('com.mapbox.mapboxsdk:mapbox-android-sdk:6.0.1') {
        transitive=true
    }
implementation 'com.mapbox.mapboxsdk:mapbox-android-core:0.2.0'
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.13.0-SNAPSHOT'
implementation('com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.13.0-SNAPSHOT') {
        transitive = true
    }
implementation 'com.mapbox.mapboxsdk:mapbox-android-plugin-locationlayer:0.5.0'
implementation 'com.google.android.gms:play-services-base:15.0.1'
implementation 'com.google.android.gms:play-services-location:15.0.1'
implementation 'com.mapzen.android:lost:3.0.4'
```

And the crash : 
```
05-03 12:24:49.612 25048-25048/com.app E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.app, PID: 25048
    java.lang.NullPointerException: Attempt to invoke interface method 'void com.mapbox.android.core.location.UpdateGoogleRequestPriority.update(com.google.android.gms.location.LocationRequest)' on a null object reference
        at com.mapbox.android.core.location.GoogleLocationEngine.updateRequestPriority(GoogleLocationEngine.java:167)
        at com.mapbox.android.core.location.GoogleLocationEngine.requestLocationUpdates(GoogleLocationEngine.java:129)
        at com.app.positionning.PositionningMapbox.onConnected(PositionningMapbox.java:147)
        at com.mapbox.android.core.location.GoogleLocationEngine.onConnected(GoogleLocationEngine.java:93)
        at com.google.android.gms.common.internal.GmsClientEventManager.onConnectionSuccess(Unknown Source:109)
        at com.google.android.gms.common.api.internal.zzav.zzb(Unknown Source:22)
        at com.google.android.gms.common.api.internal.zzaj.zzat(Unknown Source:92)
        at com.google.android.gms.common.api.internal.zzaj.onConnected(Unknown Source:21)
        at com.google.android.gms.common.api.internal.zzbd.onConnected(Unknown Source:7)
        at com.google.android.gms.common.api.internal.zzp.onConnected(Unknown Source:5)
        at com.google.android.gms.common.internal.zzf.onConnected(Unknown Source:2)
        at com.google.android.gms.common.internal.BaseGmsClient$PostInitCallback.handleServiceSuccess(Unknown Source:131)
        at com.google.android.gms.common.internal.BaseGmsClient$zza.deliverCallback(Unknown Source:62)
        at com.google.android.gms.common.internal.BaseGmsClient$zza.deliverCallback(Unknown Source:2)
        at com.google.android.gms.common.internal.BaseGmsClient$CallbackProxy.deliverCallback(Unknown Source:52)
        at com.google.android.gms.common.internal.BaseGmsClient$zzb.handleMessage(Unknown Source:270)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6938)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

Any ideas ?

**Comment by electrostat** [comment]
@Niduank  I have been unable to replicate this issue within my test apps. One thing I would look at is the `location-layer-plugin`. We have been noticing various issues with `0.5.0`. If you could test with `0.6.0-SNAPSHOT` and update here, I'd appreciate it.  We are working on a `0.6.0` release and it should be out shortly.

**Comment by ralphgabrielle** [comment]
locationEngineProvider.obtainLocationEngineBy(LocationEngine.Type.GOOGLE_PLAY_SERVICES);

returns null on 0.6.0 SNAPSHOT. Overall, is there's a way to get the users location ? I followed several Navigation tutorial by Mapbox but is not working on some device.

**Comment by electrostat** [comment]
This is being handled through this PR: https://github.com/mapbox/mapbox-events-android/pull/178

**Comment by electrostat** [comment]
@Niduank @ralphgabrielle I just pushed a new Snapshot for the Core library that should fix your problem `0.3.0-SNAPSHOT`. Give it a look and confirm it solves the crash you were seeing. I'll work on rolling this into the next available maps release.
